### PR TITLE
GSW-867 feat common calculation logic to separate package

### PR DESCRIPTION
**Pull request by r3v4s**
# New
- tick_math, liquidity_amounts calculation logic is being used by multiple realms => make it as `common` package

TODO: wait for #160 #161 to merge, then change inner code

### CODE REVIEW

**Review by mconcat — APPROVED** [review]
Removing redundant functions across packages is a great improvement, thanks for the refactoring!

**Comment by notJoon** [comment]
It's definitely became more concise, thanks for the work 👍

**Review by notJoon — DISMISSED** [review]

**Comment by r3v4s** [comment]
@notJoon @mconcat Had to force push to fix conflicts. Please take a deep look a commit cf010d148e2e8507ab8acfe7e45b4de9d14947c1

File: common/liquidity_amounts.gno [hunk 1]
```diff
@@ -9,12 +13,12 @@ func liquidityAmountsGetLiquidityForAmount0(
...
 	}
 
-	intermediate := (sqrtRatioAX96 * sqrtRatioBX96) / Q96
+	intermediate := (sqrtRatioAX96 * sqrtRatioBX96) / consts.Q96
 
 	return amount0 * intermediate / (sqrtRatioBX96 - sqrtRatioAX96)
 }
 
-func liquidityAmountsGetLiquidityForAmount1(
+func LiquidityAmountsGetLiquidityForAmount1(
```
Comment (notJoon) [inline common/liquidity_amounts.gno:21]: I think we'll have to rename it something better later.  just like truncate the `LiquidityAmount` prefix 🤔

File: README.md [hunk 2]
```diff
@@ -41,7 +41,8 @@ To run unit tests, follow these steps:
 2. Copy the `gnoswap` contracts into the Gno core:
 
    ```
-   $ cp -R _setup/* gov pool position router staker $WORKDIR/gno-for-swap/examples/gno.land/r/
+   $ cp -R _setup/* consts gov pool position router staker $WORKDIR/gno-for-swap/examples/gno.land/r/demo/
```
Comment (notJoon) [inline README.md:44]: 👍

File: staker/reward_math.gno [hunk 3]
```diff
@@ -74,7 +82,7 @@ func rewardMathComputeInternalRewardAmount(tokenId uint64, deposit Deposit) bigi
 	rewardRatio := getRewardRatio(stakedDuration)
 
 	finalRewardQ := myWholeRewardX96 * bigint(rewardRatio)
-	finalReward := finalRewardQ / Q96 / Q96 / Q96 / Q96 / Q96 / 100 / 100
+	finalReward := finalRewardQ / consts.Q96 / consts.Q96 / consts.Q96 / consts.Q96 / consts.Q96 / 100 / 100
```
Comment (notJoon) [inline staker/reward_math.gno:85]: I'm not familiar with the specific context or behind story of this code, but it appears that replacing `100 / 100` with `1000` might be a straightforward adjustment. 

Additionally, converting `blockRewardInit` to a `bigint` type seems feasible for use in this calculation, but I don't know if that variable can be used in a formula.

**Review by notJoon — DISMISSED** [review]
LGTM. I don't see any particular issues

File: common/liquidity_amounts.gno (hunk 1 above)
Comment (r3v4s) [reply common/liquidity_amounts.gno:21]: once we refactor our logic and if it works fantastic both on unit tests + integeration tests, then we can have time to rename things (including package path to deploy)

File: staker/reward_math.gno (hunk 3 above)
Comment (r3v4s) [reply staker/reward_math.gno:85]: >  but it appears that replacing 100 / 100 with 1000 might be a straightforward adjustment.

Actually, it goes same with bunch of `Q96` division. Right below that line there are some comments about why we need this kind of division formula.
https://github.com/gnoswap-labs/gnoswap/blob/64fc5ac2b64fec64e66af6bdcf96a0204b0aa105/staker/reward_math.gno#L87-L93
I kind wanted to comments to be similar as formula and this is exact reason why I didn't use `1000`. However I'm fine with your suggestion (I'll leave it you :D)

Comment (r3v4s) [reply staker/reward_math.gno:85]: > Additionally, converting blockRewardInit to a bigint type 
Suggesting, wait until @mconcat to finish his uint256 then we can do something about it. How about that?

**Review by mconcat — DISMISSED** [review]

**Review by notJoon — APPROVED** [review]

**Review by mconcat — APPROVED** [review]

**Pull request merged**